app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,30 @@
+class RestaurantsController < ApplicationController
+
+  include Response
+  include ActiveModel
+  def index
+    @restaurants = Restaurant.all
+    # Cuisine.find(@restaurants.cuisine_id)
+    # render :json => @restaurants.to_json(:except => [:created_at, :updated_at], include: 'cuisine' )
+    render :json => @restaurants.to_json(:except => [:created_at, :updated_at, :cuisine_id], :include => {:reviews => {:except => [:created_at, :updated_at, :restaurant_id]}, :cuisine => {:except => [:created_at, :updated_at]}})
+  end
+
+  def averageRating
+    @avg = nil
+    begin
+      @restaurant = Restaurant.find(params["restaurant_id"])
+      if !@restaurant.reviews.empty?
+        @avg = @restaurant.reviews.reduce(0) { |sum, el| sum + el.rating }.to_f / @restaurant.reviews.size
+      end
+    rescue ActiveRecord::RecordNotFound  => e
+      render json: { errors: e.message }, status: :bad_request
+    rescue Exception => e
```

Comment:
for a rescue-everything scenario, use rescue StandardError instead of rescue Exception
I don't remember why, but here's a link that explains it - https://thoughtbot.com/blog/rescue-standarderror-not-exception

Reply:
Also, this rescue will repeat itself for each controller
Instead, you can put it in the ApplicationController:
```
rescue_from StandardError do |exception|
  render json: { errors: e.message }, status: :internal_server_error
end
```

Reply:
I would add that in general it's a bad practice to rescue from exception in controllers, instead it should be done on the ApplicationController level via:

```ruby
rescue_from StandardError do |exception|
  response_payload = {
    error_type: exception.class.name,
    error_message: exception.message
  }

  render status: :internal_server_error,
          json: response_payload
end

rescue_from ActionController::RoutingError, ActiveRecord::RecordNotFound do |exception|
  render status: :not_found,
          json: {
            error_message: 'Resource not found'
          }
end
```